cellpack/autopack/IOutils.py

```diff
@@ -1,2292 +1,2308 @@
-                        # backward compatiblity
-                        if (
-                            ingr.o_name
-                            not in env.result_json["compartments"][orga.name][
-                                "interior"
-                            ]["ingredients"]
-                        ):
-                            continue
-                        else:
-                            name_ingr = ingr.o_name
-                    iresults, ingrname, ingrcompNum, ptInd, rad = env.getOneIngrJson(
-                        ingr,
-                        env.result_json["compartments"][orga.name]["interior"][
-                            "ingredients"
-                        ][name_ingr],
-                    )
-                    for r in iresults:
-                        rot = numpy.identity(4)
-                        if len(r[1]) == 4:  # quaternion
-                            if type(r[1][0]) == float:
-                                if transpose:
-                                    rot = tr.quaternion_matrix(
-                                        r[1]
-                                    ).transpose()  # transpose ?
-                                else:
-                                    rot = tr.quaternion_matrix(r[1])  # transpose ?
-                            else:
-                                rot = numpy.array(r[1]).reshape(4, 4)
-                            #                        ingr.results.append([numpy.array(r[0]),rot])
-                        else:
-                            rot = numpy.array(r[1]).reshape(4, 4)
-                        orgaresult[abs(ingrcompNum) - 1].append(
-                            [numpy.array(r[0]), rot, ingrname, ingrcompNum, 1]
-                        )
-    freePoint = []  # pickle.load(rfile)
-    try:
-        rfile = open(resultfilename + "freePoints", "rb")
-        freePoint = pickle.load(rfile)
-        rfile.close()
-    except:  # noqa: E722
-        pass
-    return result, orgaresult, freePoint
+# -*- coding: utf-8 -*-
+"""
+Created on Sun Jan 27 09:04:10 2013
+
+@author: Ludovic Autin
+"""
+import os
+import pickle
+
+import numpy
+from xml.dom.minidom import getDOMImplementation
+import json
+from json import JSONEncoder
+from json import encoder
+from collections import OrderedDict
+
+from numpy.core.arrayprint import printoptions
+
+import cellpack.autopack as autopack
+import cellpack.autopack.transformation as tr
+from .ingredient.Ingredient import KWDS
+from cellpack.autopack.Serializable import (
+    sCompartment,
+    sIngredientGroup,
+    sIngredient,
+    sIngredientFiber,
+)
+from cellpack.autopack.Recipe import Recipe
+from cellpack.autopack.Compartment import Compartment
+from .ingredient import (
+    ActinIngredient,
+    SingleSphereIngr,
+    MultiSphereIngr,
+    SingleCubeIngr,
+    MultiCylindersIngr,
+    GrowIngredient,
+)
+
+encoder.FLOAT_REPR = lambda o: format(o, ".8g")
+
+
+class NumpyArrayEncoder(JSONEncoder):

```

Comment:
is this one of the only changes in this file? I wonder why the diff got so messed up

Reply:
The only change here was removing line 435 and line 436. Not sure why the diff is so weird